### PR TITLE
fixes #1004 : Build image only when changes merged to master

### LINKED ISSUE #1004: `ghcr.io/containerd/nerdctl`  contains useless `dependabot-*` tags
### Description

`ghcr.io/containerd/nerdctl`  contains useless `dependabot-*` tags

https://github.com/containerd/nerdctl/pkgs/container/nerdctl/versions
image

The GHA yaml https://github.com/containerd/nerdctl/blob/master/.github/workflows/ghcr-image-build-and-publish.yml has to be fixed

### Steps to reproduce the issue

See above

### Describe the results you received and expected

Results:  `dependabot-*` tags are created
Expected: these tags should not be created

### What version of nerdctl are you using?

_No response_

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

_No response_

### Host information

_No response_

### PR DESCRIPTION
**Pull request by Rajpratik71**
fixes #1004 : Build image only when changes merged to master

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Thanks, but please sign the commit for DCO
https://github.com/apps/dco

(run `git commit -a -s --amend`, and make sure that the `Signed-off-by: NAME <EMAIL>` line with your real name is included in the commit message)

File: .github/workflows/ghcr-image-build-and-publish.yml [hunk 1]
```diff
@@ -7,11 +7,11 @@ name: Container Image Build
...
 on:
   push:
-    branches: [ ]
+    branches: [ master ]
     # Publish semver tags as releases.
     tags: [ 'v*.*.*' ]
-  pull_request:
-    branches: [ ]
+#   pull_request:
+#     branches: [ master ]
```
Comment (AkihiroSuda) [inline .github/workflows/ghcr-image-build-and-publish.yml:14]: Seems accidentally commented out?

Comment (Rajpratik71) [reply .github/workflows/ghcr-image-build-and-publish.yml:14]: No knowingly commented to doesn't built image on PR

**Comment by Rajpratik71** [comment]
> Thanks, but please sign the commit for DCO https://github.com/apps/dco
> 
> (run `git commit -a -s --amend`, and make sure that the `Signed-off-by: NAME <EMAIL>` line with your real name is included in the commit message)


Hi @AkihiroSuda, updated both the commit and PR description with Signoff

File: .github/workflows/ghcr-image-build-and-publish.yml (hunk 1 above)
Comment (AkihiroSuda) [reply .github/workflows/ghcr-image-build-and-publish.yml:14]: Can we just remove these lines?

Comment (Rajpratik71) [reply .github/workflows/ghcr-image-build-and-publish.yml]: Done as requested @AkihiroSuda

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**